File: 0000001_twoSum/php/Solution.php
<?php

class Solution {
    /**
     * @param Integer[] $nums
     * @return Integer[]
     */
    function twoSum(array $nums, int $target): array {
        $visited = [];
        foreach ($nums as $i => $num) {
            if (isset($visited[$target - $num])) {
                return [$visited[$target - $num], $i];
            }
            $visited[$num] = $i;
        }
        return [];
    }
}

File: 0000001_twoSum/java/src/main/java/Solution.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            Integer complement = target - nums[i];
            Integer previous = map.get(complement);

            if (previous != null) {
                return new int[]{previous, i};
            }

            map.put(nums[i], i);
        }

        return new int[0];
    }
}